### PR TITLE
Modify reek configuration

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
Add `.reek.yml` and remove the old file `config.reek`, which was used by `pronto`

 
## Insight
- Copy the configs from the old file
- Add some more global rules: 
  + `UncommunicativeModuleName` accepts `V1`
  + `UtilityFunction` checks `public_methods_only`
  + `DuplicateMethodCall` with { max_calls: 2 }
- Add reek comments for some custom files

## Proof Of Work
Reek should be run without any warnings

### CODE REVIEW

File: app/controllers/api/v1/application_controller.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ def current_user
       end
 
       # Render Error Message in json_api format
+      # :reek:LongParameterList { max_params: 5 }
       def render_error(detail:, source: nil, meta: nil, status: :unprocessable_entity, code: nil)
```
Comment (team-nimblehq) [inline app/controllers/api/v1/application_controller.rb:16]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/a52a836dd21ed47bf1fd4740a7355723c867b07a/app/controllers/api/v1/application_controller.rb#L16"></span>Remove unused methods (api::v1::applicationcontroller#render_error)</td>
    </tr>
  </tbody>
</table>

File: spec/support/vcr.rb [hunk 2]
```diff
@@ -39,6 +39,7 @@
     end
   end
 
+  # :reek:FeatureEnvy
   def configure_vcr_with_options(example, vcr_options)
```
Comment (team-nimblehq) [inline spec/support/vcr.rb:43]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/a52a836dd21ed47bf1fd4740a7355723c867b07a/spec/support/vcr.rb#L43"></span>Has approx 9 statements</td>
    </tr>
  </tbody>
</table>

**Comment by team-nimblehq** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/a52a836dd21ed47bf1fd4740a7355723c867b07a/app/controllers/api/v1/application_controller.rb#L10">app/controllers/api/v1/application_controller.rb#L10</a> - Remove unused methods (api::v1::applicationcontroller#current_user)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/a52a836dd21ed47bf1fd4740a7355723c867b07a/spec/support/vcr.rb#L48">spec/support/vcr.rb#L48</a> - Has the variable name ‘c’</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/controllers/api/v1/application_controller.rb (hunk 1 above)
Comment (hoangmirs) [reply app/controllers/api/v1/application_controller.rb:16]: We use it in child controllers

**Pull request merged**